subprojects/gradle/deeplink-generator/src/main/kotlin/com/avito/deeplink_generator/internal/merge/MergeDeeplinkManifestTask.kt

```diff
@@ -0,0 +1,154 @@
+package com.avito.deeplink_generator.internal.merge
+
+import com.android.build.gradle.internal.tasks.manifest.mergeManifests
+import com.android.manifmerger.ManifestMerger2
+import com.android.utils.NullLogger
+import com.avito.capitalize
+import com.avito.deeplink_generator.model.Deeplink
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.SetProperty
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+
+/**
+ * Merges default library manifest with generated manifest with intent filters
+ * to open public deeplinks.
+ */
+@CacheableTask
+internal abstract class MergeDeeplinkManifestTask : DefaultTask() {
+
+    /**
+     * Collection of public deeplinks that can launch application from outside.
+     */
+    @get:Input
+    abstract val publicLinks: SetProperty<Deeplink>
+
+    /**
+     * Fully qualified class name of an Activity responsible for handling public deeplinks.
+     */
+    @get:Input
+    abstract val activityIntentFilterClass: Property<String>
+
+    @get:InputFile
+    @get:PathSensitive(PathSensitivity.RELATIVE)
+    abstract val inputManifest: RegularFileProperty
+
+    @get:OutputFile
+    abstract val outputManifest: RegularFileProperty
+
+    @TaskAction
+    fun merge() {
+        val inputManifestFile = inputManifest.asFile.get()
+        val outputManifestFile = outputManifest.asFile.get()
+        val publicDeeplinkManifest = createPublicDeeplinkManifest(outputManifestFile.parentFile)
+        publicDeeplinkManifest.deleteOnExit()
+
+        mergeManifests(
```

Comment:
Is it an official way to use an internal implementation here? Don't we have a public API?

Reply:
Well, I didn't found any recommendations on how to merge two manifest files. 
An official example shows us that we could modify manifest https://developer.android.com/studio/build/extend-agp#supported-operations , but only as a plain text document. 

`mergeManifests` function is public, documented (a little bit) and used in gradle tasks to merge both application and library manifest. I believe, we can use it in our task as well.